client/components/data/query-atat-eligibility/index.jsx

```diff
@@ -1,43 +1,19 @@
 /**
  * External dependencies
  */
-
-import PropTypes from 'prop-types';
-import { Component } from 'react';
-import { connect } from 'react-redux';
+import React from 'react';
+import { useDispatch } from 'react-redux';
 
 /**
  * Internal dependencies
  */
 import { requestEligibility } from 'calypso/state/automated-transfer/actions';
 
-export class QueryAutomatedTransferEligibility extends Component {
-	static propTypes = {
-		requestEligibility: PropTypes.func.isRequired,
-		siteId: PropTypes.number,
-	};
-
-	UNSAFE_componentWillMount() {
-		this.request( this.props );
-	}
-
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( this.props.siteId !== nextProps.siteId ) {
-			this.request( nextProps );
-		}
-	}
+export default function QueryAutomatedTransferEligibility( siteId ) {
```

Comment:
Should `siteId` be destructered from `props` here?

And should it be a documented prop, either through TS or continuing to use PropTypes?

Reply:
Oops! I fixed the props destructuring and re-added the `siteId` proptype.

I tested the query component on a plugin upload page and it works: correctly queried the Atomic eligibility state and told me that I need a Business plan on my test site.